middleware/index.js

```diff
@@ -6,6 +6,10 @@ middlewareObj.isLoggedIn=function(req,res,next){
     If user is logged in: Redirect to next page
     else, redirect to login page
     */
+    if(req.isAuthenticated()) {
+        return next();
+    }
+    res.redirect('/login?message=You+need+to+be+logged+in+to+do+that')
```

Comment:
Again, instead of sending the error message as a parameter, use Flash to create messages which disappear after the page is viewed